### PR TITLE
*: add logs and try fixing error log

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
1. Dumpling's log is not clear when some bugs occurred.
2. We didn't check `FileRowIter.Error` when there is no next. However, we check this when we start this `rowIter`. But this is safer.

### What is changed and how it works?
1. Add more log.
2. Check Error.

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch
 
### Release note

- No release note

### CODE REVIEW

File: v4/export/dump.go [hunk 1]
```diff
@@ -241,15 +241,24 @@ func (d *Dumper) startWriters(tctx *tcontext.Context, wg *errgroup.Group, taskCh
...
 		writer.rebuildConnFn = rebuildConnFn
 		writer.setFinishTableCallBack(func(task Task) {
-			if td, ok := task.(*TaskTableData); ok {
+			if _, ok := task.(*TaskTableData); ok {
 				IncCounter(finishedTablesCounter, conf.Labels)
-				tctx.L().Debug("finished dumping table data",
-					zap.String("database", td.Meta.DatabaseName()),
-					zap.String("table", td.Meta.TableName()))
+				// FIXME: actually finishing the last chunk doesn't means this table is 'finished'.
+				//  We can call this table is 'finished' if all its chunks are finished.
```
Comment (lance6716) [inline v4/export/dump.go:247]: also rename `setFinishTableCallBack`?

File: v4/export/writer.go [hunk 2]
```diff
@@ -218,6 +220,12 @@ func (w *Writer) tryToWriteTableData(tctx *tcontext.Context, meta TableMeta, ir
 			return err
 		}
 	}
+	if somethingIsWritten {
```
Comment (lance6716) [inline v4/export/writer.go:223]:
```suggestion
	if !somethingIsWritten {
```

**Review by lance6716 — COMMENTED** [review]
rest LGTM

**Comment by hi-rustin** [comment]
@ti-chi-bot Could you please mark this pr?

**Comment by hi-rustin** [comment]
> @lichunzhu: PR needs rebase.

These reply and label will be automatically deleted after you resolve the conflict.

File: v4/export/dump.go (hunk 1 above)
Comment (lichunzhu) [reply v4/export/dump.go:247]: I don't think we need to rename `setFinishTableCallBack`. We can use a better mechanism to check whether this table is finished in the future.

**Review by lance6716 — APPROVED** [review]
/lgtm

**Pull request merged**